contract-derive/src/helpers.rs

```diff
@@ -25,13 +23,33 @@ impl<'a> From<&'a TraitItemMethod> for MethodInfo<'a> {
     fn from(method: &'a TraitItemMethod) -> Self {
         Self {
             name: &method.sig.ident,
-            args: method.sig.inputs.iter().skip(1).cloned().collect(),
+            args: method.sig.inputs.iter().cloned().collect(),
             return_type: &method.sig.output,
         }
     }
 }
 
-// Helper function to generate intercate impl from user-defined methods
+// Helper function to get the parameter names + types of a method
+pub fn get_arg_props<'a>(
+    skip_first_arg: bool,
+    method: &'a MethodInfo<'a>,
+) -> (Vec<Ident>, Vec<&syn::Type>) {
+    method
+        .args
+        .iter()
+        .skip(if skip_first_arg { 1 } else { 0 })
```

Comment:
a bit of a pedantic comment, but not a huge fan of the if here. I think having 2 different methods would be a bit cleaner

Reply:
sure, np haha

would u rather:
- create 2 completely new fns (duplicating code)
- make this fn private and create `pub fn get_arg_props_all` + `pub fn get_arg_props_skip_first` effectively exposing a cleaner api, while minimizing duplicating code
- change `skip_first_arg: bool` for `args_to_skip: u8`

Reply:
just for clarification, but we want to be able to not skip self, as it won't be an arg in the constructor

Reply:
option 2 sgtm